### PR TITLE
Update LICENSE to GPLv3

### PR DESCRIPTION
**Pull request by firefish111**
Also: I added The description and your name
image
image
As the brackets signify.

### CODE REVIEW

**Comment by firefish111** [comment]
@DynamicSquid Added submodule (you can do that)
And under the projects tab create a new project for v4, there can be brainstorming

**Comment by firefish111** [comment]
Also the correct term for writing a language in itself is bootstrapping.

**Comment by firefish111** [comment]
@DynamicSquid If interested, chcekout [this repo](http://github.com/firefish111/son)

**Comment by firefish111** [comment]
@DynamicSquid Have a look, There is this thing that links to dusk
image
When you clone the repo, it acts like a normal directory, but actually the code is stored in the OTHER repo, and cloning it clones the other repo as well

File: README.md [hunk 1]
```diff
@@ -88,11 +101,11 @@ loop for (3)
 }
 ```
 
-More information regarding the syntax can be found on the [Night website](https://night-website.dynamicsquid.repl.co/html/reference.html).
+More information regarding the syntax can be found on the [Night Website](https://night-website.dynamicsquid.repl.co/html/reference.html).
 
 ### Timeline
 
-A timeline of Night, both past and future versions. Each new release, the code is scrapped and started anew (cause I have no idea what I'm doing).
+A timeline of Night, both past and future versions. Each new release, the code is scrapped and started afresh (cause I have no idea what I'm doing).
```
Comment (DynamicSquid) [inline README.md:108]: "afresh" sounds weird, like Night is something you buy at the grocery store lol

File: README.md [hunk 2]
```diff
@@ -28,13 +41,13 @@ cd night
 With *g++*:
 
 ```
-g++ -o night src/night.cpp
+g++ -o night ./src/night.cpp
```
Comment (DynamicSquid) [inline README.md:44]: This doesnt matter

File: README.md (hunk 1 above)
Comment (firefish111) [reply README.md:108]: image

File: README.md [hunk 3]
```diff
@@ -2,24 +2,37 @@
...
 
 2. Clone this repository
 
+With *git*:
+
 ```
 git clone https://github.com/DynamicSquid/night.git
+```
+
+With *gh*: the GitHub CLI:
```
Comment (DynamicSquid) [inline README.md:27]: Bro who even uses github cli? Like I know it's cool and everything but just use git

File: README.md (hunk 1 above)
Comment (firefish111) [reply README.md:108]: so you win

**Comment by DynamicSquid** [comment]
Wait, slow down, clone the what using submodules?

File: README.md (hunk 3 above)
Comment (firefish111) [reply README.md:27]: Well the rustlings docs supply instructions for both (i think)

Comment (DynamicSquid) [reply README.md:27]: Well, a better version of github cli already exists, it's called git

**Comment by firefish111** [comment]
@DynamicSquid So you know what a submodule is right? So the act of cloning a repo with a submodule in it, does this for example:
```
gh repo clone DynamicSquid/night
# it sees there is a submodule inside .gitmodules, so it (effectively) does this:
cd night
gh repo clone firefish111/dusk # it doesn't actually do that, but that's sort of what's happening
```

File: README.md (hunk 2 above)
Comment (firefish111) [reply README.md:44]: It's good practice to

**Comment by DynamicSquid** [comment]
@firefish111 oh, okay, and so it creates a disk directory but its treated as a separate repo?

File: README.md (hunk 3 above)
Comment (firefish111) [reply README.md:27]: Actually it's the other way round but uh RUSTLINGS SUPPLIES INSTRUCTIONS FOR BOTH

**Comment by firefish111** [comment]
@DynamicSquid sortof, the thing is it doesn't have a .git inside it, that uses `..`'s `.git`. The only thing. So you can literally in your code, treat dusk/ like it is a real directory, because qhen you clone it, it will be.

**Comment by firefish111** [comment]
@DynamicSquid lol you there

**Comment by DynamicSquid** [comment]
@firefish111 so if you want to build Nigth and Dusk from source, you just have to clone this repo?

**Comment by firefish111** [comment]
@DynamicSquid á yippadeedoo.

**Comment by firefish111** [comment]
@DynamicSquid So, are we going to merge, or do we wait until v3.3?

**Comment by DynamicSquid** [comment]
Give me a sec first

**Comment by firefish111** [comment]
@DynamicSquid You've had over 1080 seconds

**Comment by DynamicSquid** [comment]
@firefish111 imma create my own license:

```
SQDLisence

You can use my code but please give credit.
```

**Comment by firefish111** [comment]
@DynamicSquid That's literally Apache, but if you add, `but if you do use it, you have to use sqdlicence too`, that's now GPL

**Comment by DynamicSquid** [comment]
@firefish111 gotcha

**Comment by firefish111** [comment]
@DynamicSquid depends on what you prefer

**Comment by DynamicSquid** [comment]
So if someone uses my code and changes the liscene to the WTF liscene, then people can copy that code and use it however?

**Comment by firefish111** [comment]
@DynamicSquid Well the thing is, they aren't allowed to change it to WTF license if you use GPL

**Comment by DynamicSquid** [comment]
So aphace allows to you do that but the gpl doesn't?

**Comment by firefish111** [comment]
@DynamicSquid Yes, another thing is GNU stuff tends to be superior

**Comment by firefish111** [comment]
@DynamicSquid also can we create a projectc for v4? I like organising things ~~because I am incredibly disorganised~~
image

**Comment by DynamicSquid** [comment]
@firefish111 "GNU is superior" is an invalid reason. *why is it superior*? Also no projects. I'm not a fan of projects on Github

**Comment by firefish111** [comment]
@DynamicSquid Oki no projects but The GNU is superior bit, here is the difference:
image
image

**Comment by CodeLongAndProsper90** [comment]
What about Projects on GitLab?

**Comment by firefish111** [comment]
@CodeLongAndProsper90 STOP STICKING YOUR NOSE INTO EVERYTHING

**Comment by DynamicSquid** [comment]
@CodeLongAndProsper90 I got it! https://gitlab.com/DynamicSquid/night

**Comment by DynamicSquid** [comment]
Ooh... it has built in CI, way better than GitHubs

**Comment by firefish111** [comment]
@DynamicSquid GitHub has builtin CI, you blonk. Check GitHub actions

**Comment by firefish111** [comment]
@DynamicSquid You know, just get get GitHub and GitLab out of the chain, and spin up a simple git server

**Comment by firefish111** [comment]
# @DynamicSquid Let's move to GitLab. I'm ready to delete dusk from here. Ok? Cool. Go ahead and delete night, and wew will have better CI from now till @CodeLongAndProsper90 starts being annoying again.

**Comment by firefish111** [comment]
I see you hhaven't got any issues or merge requetss on the GitLab repo, try these steps:

image
image

**Comment by DynamicSquid** [comment]
@firefish111 No, GitLab's only supposed to be a backup for Night. There's a reason why there aren't any Issues or Merge Requests, it's because I disabled them. Also I know that GitHub has CI, but GitLab's is way better. However I do like how GitHub supports easy CircleCI which is always nice. Also GitHub is a more popular community with a better UI in my opinion, so we're staying here :)

**Comment by firefish111** [comment]
@DynamicSquid Ok... ~~so shall I make a Night organisation here since there are now THREE repos related to night~~

**Comment by DynamicSquid** [comment]
@firefish111 Eh, no organization.

**Comment by firefish111** [comment]
@firefish111 We have firefish111/dusk, firefisk111/son, and DynamicSquid/night, it just seems too spread out...
Look at basil. Remember Basil? They have @basilTeam. rust? @rust-lang. NAme a language and I can name it's corresponding organisation.

**Comment by DynamicSquid** [comment]
@firefish111 I do not endorse SON.

**Comment by firefish111** [comment]
Hm... but technically SoN is part of Slcie and night...

Here's a problem: if you write Night in itself, it needs to be interpreted. Write an interpreter in the interpreter, the you have an interpreted interpreter which is interpreted in an interpreted interpreter, which is interpreted in an interpreted interpreter etc etc. The only REAL way for that to work is to compile it to something like SoN, so you don't get a paradox.

**Comment by DynamicSquid** [comment]
@firefish111 okay yea you right no write Night using Night (although I might write some functionality though) but no bytecode either, I want this to be a true interpreted lang

**Comment by firefish111** [comment]
@DynamicSquid You CAN do both. You can have the option of compiling it, (only for real use with the interpeter though.)

**Comment by DynamicSquid** [comment]
@firefish111 No you can't I want a interpreted lang not bytecode lang

**Comment by firefish111** [comment]
@DynamicSquid IT CAN BE BOTH

**Comment by firefish111** [comment]
@DynamicSquid you can have your cake, and eat it too

**Comment by firefish111** [comment]
WAT
image

**Comment by DynamicSquid** [comment]
@firefish111 This the the offical interpreter. You can make a compiler if you want, but it wouldn't be official

**Comment by firefish111** [comment]
Ok so that's down to two: firefish111/dusk, DynamicSquid/night, and DynamicSquid/NightWeb (hosted on github.io)

**Comment by DynamicSquid** [comment]
@firefish111 those look like honeycombs lol

**Comment by firefish111** [comment]
@DynamicSquid dey are le pumpkin

File: README.md (hunk 3 above)
Comment (firefish111) [reply README.md:27]: Actually it doesn't but I can't remember where it did

**Comment by AntimatterReactor** [comment]
Why does this turn to something far from License? Also, it's not safe to create own's license; there's a reason licenses are long as fuck. And... and i'm not going to talk again.

**Comment by DynamicSquid** [comment]
@firefish111 I added submodules to v4 branch so it's all good now

**Pull request closed without merging**